### DISCUSSION TITLE
Open-AI Model Selection - fetch available models or maybe include gpt-4o manually

### DISCUSSION THREAD

**Discussion by yourkidding** [logancyang/obsidian-copilot#463] [Ideas]
Hi there! First of all, I love your project. It really feels intuitive and can be a part of a productive workflow.

To keep up with the most up-to-date models (model names), I would love to see some kind of model selector with the ability to fetch the available model names. At the time of writing gpt-4o is the most recent model. Other Obsidian plugins just use a refresh button to get the available models with no further need to update the list of usable models.

As it is, I cannot access gpt-4o, which would bee neat because of its gigantic context window.

Thanks for your reply!

**Comment by NashP444** [comment]
Yes, please add this. I love the tool as well! I'd like to be able to use it with the most powerful and efficient GPT model
